src/shared/Core.Tests/Interop/Posix/GnuPassCredentialStoreTests.cs

```diff
@@ -89,7 +90,105 @@ public void GnuPassCredentialStore_Remove_NotFound_ReturnsFalse()
             Assert.False(result);
         }
 
-        private static string InitializePasswordStore(TestFileSystem fs, TestGpg gpg)
+        [PlatformFact(Platforms.Posix)]
+        public void GnuPassCredentialStore_ReadWriteDelete_GitRepo()
+        {
+            var trace = new NullTrace();
+            var fs = new TestFileSystem();
+            var gpg = new TestGpg(fs);
+            string storeRoot = InitializePasswordStore(fs, gpg, gitInit: true);
+
+            var mockGit = new Mock<IGit>();
+            IGit GitFactory(string path) => mockGit.Object;
+
+            var collection = new GpgPassCredentialStore(trace, fs, gpg, GitFactory, storeRoot, TestNamespace);
+
+            // Create a service that is guaranteed to be unique
+            string uniqueGuid = Guid.NewGuid().ToString("N");
+            string service = $"https://example.com/{uniqueGuid}";
+            const string userName = "john.doe";
+            const string password = "letmein123"; // [SuppressMessage("Microsoft.Security", "CS001:SecretInline", Justification="Fake credential")]
+
+            string expectedSlug = $"{TestNamespace}/https/example.com/{uniqueGuid}/{userName}.gpg";
+            string expectedFilePath = Path.Combine(storeRoot, expectedSlug);
+            string expectedFileContents = password + Environment.NewLine +
+                                          $"service={service}" + Environment.NewLine +
+                                          $"account={userName}" + Environment.NewLine;
+            byte[] expectedFileBytes = Encoding.UTF8.GetBytes(expectedFileContents);
+
+            mockGit.Setup(x => x.AddFile(expectedFilePath));
+            mockGit.Setup(x => x.Commit(It.IsAny<string>()));
+
+            try
+            {
+                // Write
+                collection.AddOrUpdate(service, userName, password);
+
+                mockGit.Verify(x => x.AddFile(expectedFilePath), Times.Once);
+                mockGit.Verify(x => x.Commit(It.IsAny<string>()), Times.Once);
+                mockGit.Reset();
+
+                // Read
+                ICredential outCredential = collection.Get(service, userName);
+
+                Assert.NotNull(outCredential);
+                Assert.Equal(userName, userName);
```

Comment:
Should we be comparing to `outcredential.Account`?

Reply:
Good